jcl/src/java.base/share/classes/java/lang/System.java

```diff
@@ -1143,6 +1151,24 @@ public static void setProperties(Properties p) {
 	}
 }
 
+static void checkTmpDir() {
+	/*[IF JAVA_SPEC_VERSION >= 20] */
+	String tmpDir = internalGetProperties().getProperty("java.io.tmpdir"); //$NON-NLS-1$
+	if (!defaultTmpDir.equals(tmpDir)) {
+		try {
+			Field systemProps = SystemProps.class.getDeclaredField("customTmpdir"); //$NON-NLS-1$
+			systemProps.setAccessible(true);
+			systemProps.set(SystemProps.class, tmpDir);
```

Comment:
The first argument can be `null`:
```java
			systemProps.set(null, tmpDir);
```
Did you consider calling `SystemProps.initProperties()` instead of using reflection here?

Reply:
I hadn't noticed the natives in `SystemProps.Raw`: Agree that can wait.